src/test/java/api/requests/scenarios/LoanDueDatesAfterRecallTests.java

```diff
@@ -525,6 +526,400 @@ public void changedDueDateAfterRecallingAnItemShouldRespectTenantTimezone()
+
+    storedLoan = loansStorageClient.getById(loan.getId()).getJson();
+    assertThat("second recall changed the due date (2 weeks)",
+        storedLoan.getString("dueDate"), is(recalledDueDate));
+  }
+
+  @Test
+  public void pagedItemRecalledThenLoanedBecomesOverdueAndNextRecallDoesNotChangeDueDate()
+      throws InterruptedException,
+      ExecutionException,
+      TimeoutException,
+      MalformedURLException {
+    final IndividualResource smallAngryPlanet = itemsFixture.basedUponSmallAngryPlanet();
+    final IndividualResource requestServicePoint = servicePointsFixture.cd1();
+    final IndividualResource steve = usersFixture.steve();
+    final IndividualResource jessica = usersFixture.jessica();
+    final IndividualResource charlotte = usersFixture.charlotte();
+
+    final LoanPolicyBuilder canCirculateRollingPolicy = new LoanPolicyBuilder()
+        .withName("Can Circulate Rolling With Recalls")
+        .withDescription("Can circulate item With Recalls")
+        .rolling(Period.weeks(3))
+        .unlimitedRenewals()
+        .renewFromSystemDate()
+        .withRecallsMinimumGuaranteedLoanPeriod(Period.weeks(2))
+        .withRecallsRecallReturnInterval(Period.weeks(1));
+
+    final IndividualResource loanPolicy = loanPoliciesFixture.create(canCirculateRollingPolicy);
+
+    useLoanPolicyAsFallback(loanPolicy.getId(),
+        requestPoliciesFixture.allowAllRequestPolicy().getId(),
+        noticePoliciesFixture.activeNotice().getId());
+
+    final IndividualResource loan = loansFixture.checkOutByBarcode(
+      smallAngryPlanet, steve, ClockManager.getClockManager().getDateTime());
+
+    final String originalDueDate = loan.getJson().getString("dueDate");
+
+    requestsFixture.placeHoldShelfRequest(smallAngryPlanet, jessica,
+        DateTime.now(DateTimeZone.UTC), requestServicePoint.getId(), "Recall");
+
+    JsonObject storedLoan = loansStorageClient.getById(loan.getId()).getJson();
+
+    final String recalledDueDate = storedLoan.getString("dueDate");
+    assertThat("due date is the original date",
+        recalledDueDate, not(originalDueDate));
+
+    final String expectedDueDate = ClockManager.getClockManager().getDateTime().plusWeeks(2).toString(ISODateTimeFormat.dateTime());
+    assertThat("due date is not the recall due date (2 weeks)",
+        storedLoan.getString("dueDate"), is(expectedDueDate));
+
+    // Move the fixed clock so that the loan is now overdue
+    ClockManager.getClockManager().setClock(Clock.offset(clock, Duration.ofDays(15)));
+
+    requestsFixture.placeHoldShelfRequest(smallAngryPlanet, charlotte,
+        DateTime.now(DateTimeZone.UTC), requestServicePoint.getId(), "Recall");
+
+    storedLoan = loansStorageClient.getById(loan.getId()).getJson();
+    assertThat("second recall changed the due date (2 weeks)",
+        storedLoan.getString("dueDate"), is(recalledDueDate));
+  }
+
+  @Test
+  public void secondRecallRequestWithMGDTruncationInPlaceDoesNotChangeDueDate()
```

Comment:
Given that the minimum duration is sooner than the recall interval after the system date, the due date shouldn't have changed even before this change was made. Is that correct?

Reply:
I'm not sure I follow. In this test the recall interval plus the system date (7 days) is sooner than the minimum guaranteed duration (14 days), so we apply the MGD to the original loan when the recall is placed. This test shows subsequent recalls do not affect the due date.

Reply:
Yes, you are right, I got the two things the wrong way round.

What I was trying to say was, if both recalls are placed within the minimum duration, then the due date wouldn't have changed even with the current code

Reply:
And I've just noticed the offset, so I'll close this

(the details of the clock freezing really needs hiding away, but that is for another day)